geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/partitions/AttributeScheme.scala

```diff
@@ -101,7 +113,9 @@ object AttributeScheme {
         val binding = sft.getDescriptor(index).getType.getBinding
         require(AttributeIndexKey.encodable(binding),
           s"Invalid type binding '${binding.getName}' of attribute '$attribute'")
-        Some(AttributeScheme(attribute, index, binding))
+        val allowedValues: Seq[String] = config.options.get(Config.AllowedListOpt).map(_.split(',').toSeq).getOrElse(Seq.empty).map(AttributeIndexKey.encodeForQuery(_,binding))
+        val defaultPartition = config.options.getOrElse(Config.DefaultPartitionOpt, AttributeIndexKey.encodeForQuery(if (allowedValues.nonEmpty) allowedValues.head else "", binding))
```

Comment:
need the encode to be applied to defaultValue too:

```suggestion
        val defaultPartition = AttributeIndexKey.encodeForQuery(config.options.getOrElse(Config.DefaultPartitionOpt, allowedValues.headOption.getOrElse("")), binding)
```